src/Shared/LayoutRenderers/AspNetResponseStatusCodeRenderer.cs

```diff
@@ -24,12 +38,23 @@ protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
                 return;
             }
 
-            int statusCode = httpResponse.StatusCode;
+            var statusCode = httpResponse.StatusCode;
+            if (statusCode < 100 || statusCode > 599)
+            {
+                // Only output valid HTTP status codes
+                return;
+            }
 
-            if (statusCode >= 100 && statusCode <= 599)
+            // .NET format strings are case-insensitive
+            // See https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings
+            if (string.Equals(Format, IntegerFormat, StringComparison.InvariantCultureIgnoreCase))
             {
                 builder.Append(statusCode);
             }
+            else
+            {
+                builder.Append(((HttpStatusCode)statusCode).ToString());
```

Comment:
Can we use `ToString(Format)` so the following Format-values are supported: `"G" or "g", "D" or "d", "X" or "x", and "F" or "f"`. See also: https://learn.microsoft.com/en-us/dotnet/standard/base-types/enumeration-format-strings

Reply:
This should be complete, I have added unit tests for each format value.

Reply:
Thank you for the extra testing of the Enum-Format-option.